react/components/NetworkStatusToast.js

```diff
@@ -0,0 +1,83 @@
+import PropTypes from 'prop-types'
+import { compose, pathOr } from 'ramda'
+import React from 'react'
+import { injectIntl, intlShape } from 'react-intl'
+import { withToast } from 'vtex.styleguide'
+
+class NetworkStatusToast extends React.Component {
+  static propTypes = {
+    hideToast: PropTypes.func.isRequired,
+    intl: intlShape.isRequired,
+    showToast: PropTypes.func.isRequired,
+    toastState: PropTypes.object.isRequired,
+  }
+
+  state = {
+    offline: false,
```

Comment:
Could this be done with hooks?

Reply:
Maybe it can... I made a class component because it is very clear about the lifecycle functions difference (`componentDidMount`, `componentDidUpdate` and `componentWillUnmount`). I think it's easier.

Reply:
Nevermind... Let's practice.